task_DEV_4/task_DEV_4/Rectangle.cs

```diff
@@ -1,40 +1,30 @@
 ﻿namespace task_DEV_4
 {
-    class Rectangle
+    class Rectangle : Square
     {
-        public int height;
-        public int width;
+        public double Height { get; set; }
+        public double Width { get; set; }        
 
-        public Rectangle()
+        public Rectangle(double height, double side) : base(side)
         {
-
-        }
-
-        public Rectangle(int height, int width)
-        {
-            this.height = height;
-            this.width = width;
-        }
-
-        public void SetRectangleParameters(int height, int width)
-        {
-            this.height = height;
-            this.width = width;
+            Height = height;
+            Width = side;
         }
 
-        public int CalculatePerimeter()
+        public void SetRectangleParameters(double height, double side)
         {
-            return (height + width) * 2;
+            Height = height;
+            Width = side;
         }
 
-        public int CalculateArea()
+        public override double CalculatePerimeter()
         {
-            return height * width;
+            return (Height + Width) * 2;
```

Comment:
You need to do smth like Height * Side. Where Side is property of Square and Height is property of rectangle.

Reply:
Done